lib/facter/framework/core/fact_loaders/fact_loader.rb

```diff
@@ -20,8 +20,12 @@ def initialize
 
     def load(user_query, options)
       @internal_facts = load_internal_facts(user_query, options)
-      @custom_facts = load_custom_facts(options)
-      @external_facts = load_external_facts(options)
+      if ENV['INSIDE_FACTER']
+        @log.debug('INSIDE_FACTER env var detected, not loading custom/external facts off to prevent recursion')
```

Comment:
Could you include a warning message that tells users to specify the `--no-external-facts` option like Facter 3 did https://github.com/puppetlabs/facter/blob/6d7ffc6efdfbc3b1fc79311cdeb4581ac2098d9c/exe/facter.cc#L286?

Reply:
oh I think this part was missed in the last commit. I think it'd be good to continue warning folks (like Facter 3) that they need to fix their external facts.